File: 02_Object_Oriented_Programming/02_Class_Relationships/06_Realization/java/CashPayment.java
/**
 * Realization — CashPayment implements Payment.
 */
public class CashPayment implements Payment {

    @Override
    public void pay(double amount) {
        System.out.println("Paid $" + amount + " using Cash.");
    }

    @Override
    public String getPaymentMethod() {
        return "Cash";
    }
}

File: 02_Object_Oriented_Programming/02_Class_Relationships/06_Realization/python/cash_payment.py
"""
Realization — CashPayment implements Payment.
"""

from payment import Payment


class CashPayment(Payment):
    """Pays using cash."""

    @property
    def payment_method(self) -> str:
        return "Cash"

    def pay(self, amount: float) -> None:
        print(f"Paid ${amount} using Cash.")
